application/src/main/java/org/togetherjava/tjbot/commands/system/BotCore.java

```diff
@@ -94,54 +99,121 @@ public BotCore(JDA jda, Database database, Config config) {
 
+    /**
+     * Validates the given interactor
+     *
+     * @return whenever the command should be checked
+     */
+    @NotNull
+    private static Predicate<UserInteractor> validateInteractor() {
+        return interactor -> {
+            String name = interactor.getName();
+
+            if (name == null) {
+                logger.error("An interactor's name shouldn't be null! {}", interactor);
+                return false;
+            }
+
+            if (name.startsWith("s-") || name.startsWith("mc-") || name.startsWith("uc-")) {
+                logger.warn(
+                        "Names that starts with s-, mc- or uc- aren't recommended and might break other commands. (name: {})",
+                        name);
+            }
+
+            return true;
+        };
+    }
+
     @Override
-    public Collection<SlashCommand> getSlashCommands() {
-        return nameToInteractor.values()
-            .stream()
-            .filter(SlashCommand.class::isInstance)
-            .map(SlashCommand.class::cast)
-            .toList();
+    @Unmodifiable
+    public Collection<UserInteractor> getInteractors() {
+        return interactors;
     }
 
     @Override
-    public Optional<SlashCommand> getSlashCommand(String name) {
-        return Optional.ofNullable(nameToInteractor.get(name))
-            .filter(SlashCommand.class::isInstance)
-            .map(SlashCommand.class::cast);
+    public Optional<UserInteractor> getInteractor(final String name) {
+
+        return Optional.ofNullable(nameToInteractor.get(name));
+    }
+
+    public <T extends UserInteractor> Optional<T> getInteractor(final String name,
+            @Nullable final Class<? extends T> type) {
+        String prefix = "";
+
+        if (null == type) {
+            List<UserInteractor> anyInteractors = getAnyInteractors(name);
+            if (anyInteractors.isEmpty()) {
+                return Optional.empty();
+            } else {
+                return Optional.ofNullable((T) anyInteractors.get(0));
+            }
+        } else if (SlashCommand.class.isAssignableFrom(type)) {
+            prefix = "s-";
+        } else if (MessageContextCommand.class.isAssignableFrom(type)) {
+            prefix = "mc-";
+        } else if (UserContextCommand.class.isAssignableFrom(type)) {
+            prefix = "uc-";
+        }
+
+        return Optional.ofNullable(nameToInteractor.get(prefix + name))
+            .filter(type::isInstance)
+            .map(type::cast);
     }
 
+    public List<UserInteractor> getAnyInteractors(final String name) {
+
+        List<UserInteractor> localInteractors = new ArrayList<>(4);
+
+        getInteractor(name, SlashCommand.class).ifPresent(localInteractors::add);
+        getInteractor(name, MessageContextCommand.class).ifPresent(localInteractors::add);
+        getInteractor(name, UserContextCommand.class).ifPresent(localInteractors::add);
+        getInteractor(name, UserInteractor.class).ifPresent(localInteractors::add);
```

Comment:
This feels fragile. If we add more, we would definitely forgot to adjust this method. This should be done in a more automatic way instead

Reply:
Well, its important to not just ship "working code just to unblock people" but to ship proper code.
We are not in a hurry, there is no need to ship bad stuff. We can take our time to make it nice.

If you do not want to put in more work/time, that is totally fine as well. Others can always take over if you want.

Reply:
This has been reworked to use the enum